### PR TITLE
[12.0] [MIG] sale_order_line_sequence

### PR DESCRIPTION
**Pull request by nikul-serpentcs**
Sale Order Line Sequence
========================

- Displays the sequence of Sale order line and helps to maintain the order.
- The line sequence number is also displayed in sale order reports.

### CODE REVIEW

**Comment by Swapnesh-SerpentCS** [comment]
@nikul-serpentcs 
Please check travis

**Comment by nikul-serpentcs** [comment]
@Swapnesh-SerpentCS Now `Mr. Travis` happy :smile: 

@pedrobaeza @JayVora-SerpentCS Could you please review it.

**Comment by JayVora-SerpentCS** [comment]
Runbot fails, plz check.

**Comment by nikul-serpentcs** [comment]
@JayVora-SerpentCS I don't know why `runbot` is fails, but here works fine: http://3347726-744-38f351.runbot3.odoo-community.org/web#action=257&model=sale.order&view_type=list&menu_id=138

**Comment by pedrobaeza** [comment]
This is the warning on runbot:
```
2018-11-26 07:37:22,776 153 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (sequence2, sequence) of sale.order.line() have the same label: Sequence.
```

File: sale_order_line_sequence/README.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/sale-workflow/issues>`_.
```
Comment (sudhir-erpharbor) [inline sale_order_line_sequence/README.rst:39]: version: 12.0

File: sale_order_line_sequence/README.rst [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+.. |badge1| image:: https://img.shields.io/badge/maturity-Beta-yellow.png
+    :target: https://odoo-community.org/page/development-status
+    :alt: Beta
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fsale--workflow-lightgray.png?logo=github
+    :target: https://github.com/OCA/sale-workflow/tree/11.0/sale_order_line_sequence
```
Comment (sudhir-erpharbor) [inline sale_order_line_sequence/README.rst]: Change version 11.0  to 12.0 in the file

File: sale_order_line_sequence/README.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+.. image:: https://odoo-community.org/logo.png
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+This module is part of the `OCA/sale-workflow <https://github.com/OCA/sale-workflow/tree/11.0/sale_order_line_sequence>`_ project on GitHub.
```
Comment (sudhir-erpharbor) [inline sale_order_line_sequence/README.rst]: 12.0

**Review by sudhir-erpharbor — COMMENTED** [review]
Small Changes.

File: sale_order_line_sequence/README.rst (hunk 1 above)
Comment (nikul-serpentcs) [reply sale_order_line_sequence/README.rst:39]: @sudhir-erpharbor  I think here @OCA-git-bot change (Weblate) automatically, So no need.

CC @pedrobaeza @aheficent

**Comment by nikul-serpentcs** [comment]
@JayVora-SerpentCS @pedrobaeza `Travis & Runbot` are Done.

File: sale_order_line_sequence/README.rst (hunk 3 above)
Comment (alexey-pelykh) [reply sale_order_line_sequence/README.rst]: @nikul-serpentcs seems that README was not regenerated using `oca-gen-addon-readme`

File: sale_order_line_sequence/views/report_saleorder.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
```
Comment (alexey-pelykh) [inline sale_order_line_sequence/views/report_saleorder.xml:2]: Please add license/copyright

File: sale_order_line_sequence/views/sale_view.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
```
Comment (alexey-pelykh) [inline sale_order_line_sequence/views/sale_view.xml:3]: Same here, please add license/copyright

**Review by alexey-pelykh — APPROVED** [review]
Few non-functional changes, otherwise code LGTM

File: sale_order_line_sequence/views/report_saleorder.xml (hunk 4 above)
Comment (nikul-serpentcs) [reply sale_order_line_sequence/views/report_saleorder.xml:2]: @alexey-pelykh It's not a mandatory one in XML, no need here

File: sale_order_line_sequence/views/sale_view.xml (hunk 5 above)
Comment (nikul-serpentcs) [reply sale_order_line_sequence/views/sale_view.xml:3]: same here

File: sale_order_line_sequence/views/report_saleorder.xml (hunk 4 above)
Comment (alexey-pelykh) [reply sale_order_line_sequence/views/report_saleorder.xml:2]: Absolutely true, that's why not a request change yet just a comment. Still a code, of some sort, still licensed somehow

**Comment by nikul-serpentcs** [comment]
@alexey-pelykh @sudhir-erpharbor Improved As per comment.

@sudhir-erpharbor Could you Please review it?

File: sale_order_line_sequence/README.rst [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+.. contents::
+   :local:
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/sale-workflow/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed
+`feedback <https://github.com/OCA/sale-workflow/issues/new?body=module:%20sale_order_line_sequence%0Aversion:%2011.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (sudhir-erpharbor) [inline sale_order_line_sequence/README.rst]: Still link is contains version 11.0
Change it to 12.0 in the feedback link (version: 11.0)
?body=module:%20sale_order_line_sequence%0Aversion:%2011.0

**Comment by nikul-serpentcs** [comment]
@sudhir-erpharbor Code done as per Comments.

**Review by JagdishPanchalSCS — APPROVED** [review]

**Comment by nikul-serpentcs** [comment]
@JayVora-SerpentCS @pedrobaeza @sudhir-erpharbor Commit Squashed and Travis Done.

Can you please check & move forward?

**Comment by sudhir-erpharbor** [comment]
LGTM

**Comment by nikul-serpentcs** [comment]
@sudhir-erpharbor thanks! please change the review status from "Changes Requested" to "Approved" if LGTM +1

**Review by sudhir-erpharbor — APPROVED** [review]

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

File: sale_order_line_sequence/model/sale_order.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        help="Gives the sequence of this line when displaying the sale order.",
+        default=9999,
+        string="Sequence"
+    )
+
+    # displays sequence on the order line
+    sequence2 = fields.Integer(
+        help="Shows the sequence of this line in the sale order.",
+        related='sequence',
+        string="Sequence 2",
```
Comment (AaronHForgeFlow) [inline sale_order_line_sequence/model/sale_order.py]: Can you please change this string to something less ugly? This appears in the SO view.

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
It works fine. Just one discrepancy.

File: sale_order_line_sequence/model/sale_order.py (hunk 7 above)
Comment (nikul-serpentcs) [reply sale_order_line_sequence/model/sale_order.py]: @aheficent Yes I know, But if name is same then runbot get warning like this https://github.com/OCA/sale-workflow/pull/744#issuecomment-441557570

That's why I put name `Sequence 2`.

Comment (AaronHForgeFlow) [reply sale_order_line_sequence/model/sale_order.py]: Can you put in the sequence field then `string="Hidden sequence"` ? Then you will be able to put sequence in the other

Comment (nikul-serpentcs) [reply sale_order_line_sequence/model/sale_order.py]: @aheficent Done

Comment (AaronHForgeFlow) [reply sale_order_line_sequence/model/sale_order.py]: Sorry I didn't explain it correctly. I meant to change the string of the sequence field. But I realize now it was also a bad idea.

I think is better what you did but with a more intuitive string. Can you put something like:

```
     sequence2 = fields.Integer(
        help="Shows the sequence of this line in the sale order.",
        related='sequence',
        string="Line Number",
        readonly=True,
        store=True
    )
```

**Comment by nikul-serpentcs** [comment]
@aheficent Improved code as per comment.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1:  Thanks

**Pull request merged**